### DISCUSSION TITLE
Pocketbase JS SDK & Auth.js Integration misforming http requests from the same client?

### DISCUSSION THREAD

**Discussion by tajkirkpatrick** [pocketbase/pocketbase#1669] [Q&A] [answered]
Hello everyone!

I am attempting to create a Pocketbase [adapter](https://authjs.dev/guides/adapters/using-a-database-adapter) for the new framework-agnostic core of the [Auth.js](https://github.com/nextauthjs/next-auth) library. Thankfully, there are a good amount of [resources](https://github.com/nextauthjs/next-auth/tree/main/packages) avaliable, and a future pocketbase adapter would resemble an existing prisma or supabase auth.js adapter. I've assembled somewhat of a functional adapter although, I am admittedly stuck now and seeking help from anyone with more knowledge of the pocketbase js sdk.

Problem:
Auth.js offers a jest testing suite to evaluate WIP adapters. The suite includes function calls that are calls to the database sdk saving the user's authenticated credentials, session token, cookie data, etc...

So far I've achieved the results below from the test suite:
<img width="927" alt="Screenshot 2023-01-22 at 5 39 04 PM" src="https://user-images.githubusercontent.com/50558516/213947502-ef0ce7ff-c7ea-4711-9c96-e0cc9f82f99a.png">

My specific error is a 404 that a resource can't be found (because the resource was never created), due to a missed fetch call. I have checked the pocketbase logs from the Admin UI perspective, but how can I change a misformatted fetch from within the pocketbase sdk?

<img width="919" alt="Screenshot 2023-01-22 at 6 17 42 PM" src="https://user-images.githubusercontent.com/50558516/213948341-a6e7e7bd-b179-4e9d-868b-f89e41e90468.png">

<img width="1366" alt="Screenshot 2023-01-22 at 6 20 55 PM" src="https://user-images.githubusercontent.com/50558516/213948460-0361e4ca-4243-4012-aca6-4830b8690040.png">

As you can see, the jest suite recognizes that my adapter file has the required functions methods defined, can successfully create and update the `sessions` collection and records, as well as the `verificationToken` collection (the test suite doesn't detect a delete from this collection, but I may be implementing the test's solution incorrectly). These records are confirmed in the Pocketbase Admin UI, confirming that the jest suite is sending SOME correctly configured sdk calls, although any `create` pocketbase sdk calls to the `user` or `account` tables are returning a 404 client error. 

If you're intrigured by the issue so far I can offer more details:

Steps to reproduce:
1. Fork my example repo, found [here](https://github.com/tajkirkpatrick/authjs-astro/tree/feat/adapter-pocketbase)
- The root directory of the repo already includes a minimal `pb_migrations` folder, to ensure we're working within the same environment (ty pocketbase).

2. Place the latest Pocketbase [executable](https://pocketbase.io/docs/) for your machine within the root directory. Next to `pb_migrations`. Run `./pocketbase(.exe) (--)serve` for your machine. Login to pocketbase, and the tables should appear automatically.

2. Change directories from the root of the project directory into `./packages/adapter-pocketbase`
3. Install dependencies with pnpm 
- The adapter is located within a monorepo, so pnpm will install from the scope and dependencies of the 27 packages in the repo. I know, a lot, but this ensure the adapter remains cross-platform between the Next.js framework and the new framework-agnostic core.
```sh 
pnpm install 
```

4. Lastly, from within `<root>/packages/adapter-pocketbase` run 
```sh
pnpm test
``` 
and jest should run the test suite.

I have done some hacking around already, like you'll see with the jest polyfill of fetch(). I am new to contributing to OSS so please let me know if I've missed something obvious and teachable. Thank you!

**Comment by tajkirkpatrick** [comment, marked as answer]
I've started passing the test provided by Auth.js but have encountered a different error about the crypto api within @auth/core so I'll close this discussion for now.

**Reply by leikoilja** [reply]
@tajkirkpatrick have you had any luck creating a pocketbase integration?

**Reply by tajkirkpatrick** [reply]
@leikoilja 
tl;dr : yes I made significant progress here is a link to the PR.
https://github.com/nextauthjs/next-auth/pull/6498

Yes, I made a ton of progress including getting all of the Authjs tests to pass (except for two unimplemented methods by AuthJS). Life happened so I found a stopping point and haven't picked it back up, and this was my first contribution to OSS. So don't expect best practices. Follow the thread in the PR for more information and let me know if you have any questions or suggestions to improve the adapter.

**Comment by AriaFantom** [comment]
And update ?

**Comment by TheOrdinaryWow** [comment]
And update plz?

**Comment by 0xsarwagya** [comment]
Our Team Has Created An Adapter Recently Since We Are Using Pocketbase Now.

It's Still Under Active Development and Contributions are Welcome

[You Can look at The Github Repo Here](https://github.com/Rebackk-Team/pocketbase-adapter)
[Also The NPM Package Here](https://www.npmjs.com/package/@rebackk/pocketbase-adapter)

**Reply by kyds3k** [reply]
@0xsarwagya what happened to your Github repo? Is this still in development?
